IBM/FHIR
# Support "special" search param Location-near #259

Support "special" search param Location-near #259

This PR is broken into 3 core elements: 
1 - fhir-search location enablement 
2 - fhir-persistence-jdbc location enablement
3 - extraction of parameter values

    - Add support for 'special' in parseQueryString with Prefix
    - Add near.unitTest.properties to turn on fhir-search and
    fhir-persistence tracing
    - Update the AbstractQueryBuilder to wrap location parsing errors in a
    FHIRPersistenceException and re-throw
    - Add configuration for FHIRConfiguration for switching bounding type
    - Update the LocationParmBehaviorUtil to use BoundingType configuration
    - Add JDBCSearchNearTest
    - Update Units-of-Measure
    - Update FHIR Search NearLocationHandler
    - Update Bounding interface with Box, Radius types
    - Extend tests to cover the new types, and confirm validity of the types
    - Introduce enumeration with BoundingType
    - Update the QueryBuilder signature
    - Update NearLocationHandler
    - Refactor Units to use units-of-measure definitions
            - StatuteLengths
            - ImperialUnits
            - Metric Units
            - Remove OtherUnits (Angstrom)
            - configure Units in lookup map in UOMManager
    - Refactor FHIRPersistenceUtil to put NEAR processing into fhir-search
    - AbstractQueryBuilder
            - Change the boundingbox to a list of boundingradius
            - Change to checking near (near-distance is deprecated) and add to a
    check to NearLocationHandler to support a generic check
            - Add Bounding, BoundingBox, BoundingRadius
            - Add Tests for Bounding types
    - Update QuerySegmentAggregator to use the single check for near
    - Update JDBCQueryBuilder to match the changes to AbstractQueryBuilder
    - Introduce LocationBehaviorUtil to more consistently model the Location
    signature query
    - Remove from ParameterDAOImplementation near-distance
    - Add SearchConstants.Type 'special'
    - Add JDBCSearchNearTest with 'near'
    - Remove unused call to continue in SearchUtil
    - Update the validation for the BoundingBox
    - Remove the guard for the 'special' type which blocks near processing
    - Add support for the extraction routine in the JDBC visitor code
    (Location.position)
    - Add logicalId to the ParameterVisitorBatchDAO location
    - Add 'special' to the DTO Parameter object
